netcore/System.Private.CoreLib/src/System/Threading/Thread.Mono.cs

```diff
@@ -294,7 +294,13 @@ ThreadState ValidateThreadState ()
 		extern static void InitInternal (Thread thread);
 
 		[MethodImplAttribute(MethodImplOptions.InternalCall)]
-		extern static Thread InitializeCurrentThread ();
+		private extern static void InitializeCurrentThread_icall (ref Thread thread);
+
+		static Thread InitializeCurrentThread () {
```

Comment:
We are supposed to mark this NoInline right?

Reply:
Hm. yea. I think we want to try to replace it with an intrinsic in the near future